### PR TITLE
Update

### LINKED ISSUE #169: missing compatibility with h11 0.13.0
building against newer h11 releases gives these test failures:

```
[   33s] =================================== FAILURES ===================================
[   33s] ________________________________ test_handshake ________________________________
[   33s] 
[   33s]     def test_handshake() -> None:
[   33s]         response, nonce = _make_handshake([])
[   33s]     
[   33s] >       response.headers = sorted(response.headers)  # For test determinism
[   33s] 
[   33s] test/test_server.py:208: 
[   33s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   33s] 
[   33s] self = InformationalResponse(headers=<Headers([(b'upgrade', b'WebSocket'), (b'connection', b'Upgrade'), (b'sec-websocket-accept', b'h8YJtSXCJagtWfHvVj0GxvhYXrM=')])>, http_version=b'1.1', reason=b'', status_code=101)
[   33s] name = 'headers'
[   33s] value = [(b'connection', b'Upgrade'), (b'sec-websocket-accept', b'h8YJtSXCJagtWfHvVj0GxvhYXrM='), (b'upgrade', b'WebSocket')]
[   33s] 
[   33s] >   ???
[   33s] E   dataclasses.FrozenInstanceError: cannot assign to field 'headers'
[   33s] 
[   33s] <string>:4: FrozenInstanceError
[   33s] _________________________ test_handshake_extra_headers _________________________
[   33s] 
[   33s]     def test_handshake_extra_headers() -> None:
[   33s]         response, nonce = _make_handshake([], accept_headers=[(b"X-Foo", b"bar")])
[   33s]     
[   33s] >       response.headers = sorted(response.headers)  # For test determinism
[   33s] 
[   33s] test/test_server.py:222: 
[   33s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   33s] 
[   33s] self = InformationalResponse(headers=<Headers([(b'upgrade', b'WebSocket'), (b'connection', b'Upgrade'), (b'sec-websocket-accept', b'sbk5BhBG0603A9oxAGV4giGOq9k='), (b'x-foo', b'bar')])>, http_version=b'1.1', reason=b'', status_code=101)
[   33s] name = 'headers'
[   33s] value = [(b'connection', b'Upgrade'), (b'sec-websocket-accept', b'sbk5BhBG0603A9oxAGV4giGOq9k='), (b'upgrade', b'WebSocket'), (b'x-foo', b'bar')]
[   33s] 
[   33s] >   ???
[   33s] E   dataclasses.FrozenInstanceError: cannot assign to field 'headers'
[   33s] 
[   33s] <string>:4: FrozenInstanceError
```

### PR DESCRIPTION
**Pull request by pgjones**
H11, and python versions

fixes #169

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -31,23 +31,22 @@
...
-        'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
         'Programming Language :: Python :: Implementation :: CPython',
         'Programming Language :: Python :: Implementation :: PyPy',
     ],
     install_requires=[
-        "dataclasses ; python_version < '3.7'",
```
Comment (Kriechi) [inline setup.py:50]: Is there a specific incompatibility with dataclasses on 3.6 that makes it easier to simply drop 3.6 and move to "native" 3.7?

Comment (pgjones) [reply setup.py:50]: dataclasses was added to the stdlib in 3.7

Comment (Kriechi) [reply setup.py:50]: yes, my question was actually: why drop support for 3.6?
or was this needed to fix the compatibility with h11?

Comment (pgjones) [reply setup.py:50]: oh right. 3.6 is now unsupported, https://endoflife.date/python. h11 will drop 3.6 in the next release as well.

**Pull request merged**